src/util/fields.py

```diff
@@ -147,7 +147,12 @@ def clean_context(context):
         [model, r"\y{}\y".format(fieldname)],
     )
     for id_, name, context_s in cr.fetchall():
-        context = safe_eval(context_s or "{}", SelfPrintEvalContext(), nocopy=True)
+        try:
+            context = safe_eval(context_s or "{}", SelfPrintEvalContext(), nocopy=True)
+        except SyntaxError:
+            cr.execute("DELETE FROM ir_filters WHERE id = %s", [id_])    
+            add_to_migration_reports(("ir.filters", id_, name), "Filters/Dashboards")
```

Comment:
The report actually says that the filter has been corrected.
It also adds a 404 link to the removed record.
Also, why removing the filter? I would simply skip it (with a `logger.warning`).